lib/spack/spack/environment.py

```diff
@@ -1472,13 +1480,18 @@ def write(self, regenerate_views=True):
 
         # Remove yaml sections that are shadowing defaults
         # construct garbage path to ensure we don't find a manifest by accident
-        bare_env = Environment(os.path.join(self.manifest_path, 'garbage'),
-                               with_view=self.view_path_default)
-        keys_present = list(yaml_dict.keys())
-        for key in keys_present:
-            if yaml_dict[key] == config_dict(bare_env.yaml).get(key, None):
-                if key not in raw_yaml_dict:
-                    del yaml_dict[key]
+        env_dir = None
+        try:
+            env_dir = tempfile.mkdtemp()
+            bare_env = Environment(env_dir, with_view=self.view_path_default)
+            keys_present = list(yaml_dict.keys())
+            for key in keys_present:
+                if yaml_dict[key] == config_dict(bare_env.yaml).get(key, None):
+                    if key not in raw_yaml_dict:
+                        del yaml_dict[key]
+        finally:
+            if env_dir:
+                shutil.rmtree(env_dir)
```

Comment:
Can you replace the above with:

```python
with fs.tmp_cwd() as env_dir:
    # do stuff
```

It should handle the `mkdtemp` cruft (if it doesn't, maybe modify it to be more permissive -- I think it is only lacking your `if env_dir`)